src/main/java/com/faforever/server/geoip/GeoIpService.java

```diff
@@ -63,10 +64,30 @@ public void updateDatabaseFile() throws IOException {
   }
 
   public Optional<String> lookupCountryCode(InetAddress inetAddress) {
+    Optional<CityResponse> city = lookupCity(inetAddress);
+    return city.isPresent() ? Optional.of(city.get().getCountry().getIsoCode()) : Optional.empty();
+  }
+
+  /**
+   * @return Time zone code as specified by the IANA
+   */
+  public Optional<String> lookupTimezone(InetAddress inetAddress) {
+    Optional<CityResponse> city = lookupCity(inetAddress);
+
+    if (!city.isPresent()) {
+      return Optional.empty();
+    } else {
+      return noCatch(() -> {
+        return Optional.ofNullable(city.get().getLocation().getTimeZone());
+      });
+    }
```

Comment:
Does anybody have a better way of doing this Optional.ifPresent(return).else(return)?